hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeManager.java

```diff
@@ -1189,16 +1190,26 @@ public void registerDatanode(DatanodeRegistration nodeReg)
             nodes with its data cleared (or user can just remove the StorageID
             value in "VERSION" file under the data directory of the datanode,
             but this is might not work if VERSION file format has changed 
-         */        
+         */
+          // Check if nodeS's host information is same as nodeReg's, if not,
+          // it needs to update host2DatanodeMap accordringly.
+          updateHost2DatanodeMap = !nodeS.getIpAddr().equals(nodeReg.getIpAddr()) ||
+                  (nodeS.getXferPort() != nodeReg.getXferPort());
+
```

Comment:
Can we update 
`updateHost2DatanodeMap = !nodeS.getIpAddr().equals(nodeReg.getIpAddr()) ||
                  (nodeS.getXferPort() != nodeReg.getXferPort());`
to
`updateHost2DatanodeMap = !nodeS.getXferAddr().equals(nodeReg.getXferAddr());`

Reply:
Makes sense as getXferAddr() is "getIpAddr():getXferPort()".

Reply:
Updated.